File: pset6/readability/readability.py
# program to determine readability of text
from cs50 import get_string


text = get_string("Text: ")
length = len(text)
letters = 0.0
words = 1.0
sentences = 0.0

# going through each character 1 by 1
# to determine if its letter, word, sentence
for i in range(length):
    if text[i].isalpha():
        letters += 1
    elif text[i] == " ":
        words += 1
    elif text[i] in [".", "!", "?"]:
        sentences += 1
        
L = letters / words * 100
S = sentences / words * 100
x = round(0.0588 * L - 0.296 * S - 15.8)
    
if x < 1:
    print("Before Grade 1")
elif x > 15:
    print("Grade 16+")
else:
    print("Grade " + str(x))

File: pset2/readability/readability.c
#include <stdio.h>
#include <math.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

int main(void)
{
    string text;
    int stringlen;
    float letters = 0.0, words = 1.0, sntc = 0.0;
    float grade, L, S;
    
    text = get_string("Text: ");
    stringlen = strlen(text);
    
    for (int i = 0; i < stringlen; i++)
    {
        if (isalpha(text[i]))
        {
            letters += 1;
        }
        if (isblank(text[i]))
        {
            words += 1;
        }
        if (text[i] == 33 || text[i] == 46 || text[i] == 63)
        {
            sntc += 1;
        }
    }
    
    L = (letters / words) * 100;
    S = (sntc / words) * 100;
    grade = round(0.0588 * L - 0.296 * S - 15.8);
    
    if (grade < 1)
    {
        printf("Before Grade 1\n");
        return 0;
    }
    if (grade >= 16)
    {
        printf("Grade 16+\n");
        return 0;
    }
    printf("Grade %.0f\n", grade);
    return 0;
}